Bowling_Calculator/BowlingCalculator.cs

```diff
@@ -0,0 +1,283 @@
+            {
+                AddFrameScoreToFrame(frame_id_counter - 2, result);
+                UpdateTotalScoreOfFrame(frame_id_counter - 2);
+                results[frame_id_counter - 1].Add_resultThrow(result);
+
+
+                if (results[frame_id_counter - 1].Get_isFrameEnded())
+                {
+                    AddAllScores();
+                }
+                else
+                {
+                    int previous_total = results[frame_id_counter - 2].Get_total();
+                    results[frame_id_counter - 1].Set_total(previous_total + result);
+                }
+            }
+            // 3d or more frame has been started
+            else if ((frame_id_counter > 2) && (frame_id_counter < 10))
+            {
+                AddFrameScoreToFrame(frame_id_counter - 3, result);
+                AddFrameScoreToFrame(frame_id_counter - 2, result);
+                UpdateTotalScoreOfFrame(frame_id_counter - 3);
+                UpdateTotalScoreOfFrame(frame_id_counter - 2);
+                results[frame_id_counter - 1].Add_resultThrow(result);
+
+                if (results[frame_id_counter - 1].Get_isFrameEnded())
+                {
+                    AddAllScores();
+                }
+                else
+                {
+                    int previous_total = results[frame_id_counter - 2].Get_total();
+                    results[frame_id_counter - 1].Set_total(previous_total + result);
+                }
+            }
+            // 10th frame has been started
+            else if (frame_id_counter == 10)
+            {
+                AddFrameScoreToFrame(frame_id_counter - 3, result);
+                AddFrameScoreToFrame(frame_id_counter - 2, result);
+                UpdateTotalScoreOfFrame(frame_id_counter - 3);
+                UpdateTotalScoreOfFrame(frame_id_counter - 2);
+                results[frame_id_counter - 1].Set_isBonusThrow(true);
+
+                results[frame_id_counter - 1].Add_resultThrow(result);
+                UpdateTotalScoreOfFrame(frame_id_counter - 1);
+            }
+        }
+
+        public List<int> Get_frame_result(int frame_id)
+        {
+            return results[frame_id - 1].Get_resultThrows();
+        }
+
+        public int Get_frame_score(int frame_id)
+        {
+            return results[frame_id - 1].Get_frameScore();
+        }
+
+        public int Get_total(int frame_id)
+        {
+            return results[frame_id - 1].Get_total();
+        }
+
+        public int Get_latest_total_score()
+        {
+            return results[results.Count - 1].Get_total();
+        }
+
+        class Frame
+        {
+            int id;
+            bool isStrike;
+            bool isSpare;
+            bool isBonusThrow;
+            bool isFrameEnded;
+            List<int> resultThrows = new List<int>();
+            List<int> addedThrows = new List<int>();
+            int frameScore;
+            int total;
```

Comment:
All this fields is private so them names should start with `_` (underscore) by naming convention. Fix please.